### ISSUE TITLE
Scala: Taint mode matching issue

### ISSUE THREAD

**Issue by inkz** [semgrep/semgrep#4779]
While trying to highlight code inside web frameworks controller (`Action => {...}`) taint mode work only in one situation:
https://semgrep.dev/s/A5Jw

```yaml
rules:
- id: tainted-html-response
  mode: taint
  pattern-sinks:
  - patterns:
    - pattern: Ok(...)
  pattern-sources:
  - patterns:
    - pattern: $PARAM
    - pattern-inside: |
        def $METHOD(..., $PARAM: $TYPE, ...) = Action {
          ...
        }
  message: test123
  languages: [scala]
  severity: WARNING
```

```scala
package controllers

import play.api.mvc._
import javax.inject._
import models.UserModel

@Singleton
class AuthController @Inject()(val controllerComponents: ControllerComponents) extends BaseController{
  
  def test1(input: String) = Action { implicit request: Request[AnyContent] =>
    // working :)
    Ok(input)
  }

  def test2(input: String) = Action {
    // not working :(
    Ok(input)
  }
}

```

Search mode works fine:

https://semgrep.dev/s/BG36

```yaml
rules:
- id: test123
  patterns:
    - pattern: Ok(...)
    - pattern-inside: |
        Action {
          ...
        }
  message: test123
  languages: [scala]
  severity: WARNING
```

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-947/scala-taint-mode-matching-issue.

**Comment by inkz** [comment]
One more example: https://semgrep.dev/s/Og3R

**Comment by aryx** [comment]
Does this work now @IagoAbal ?

**Comment by IagoAbal** [comment]
Yeah, my bad, this should have been closed by #4792.
